docker/freebsd.sh

```diff
@@ -67,21 +111,21 @@ latest_freebsd() {
 }
 
 freebsd_mirror() {
-    local home=
+    local mirror=
     local code=
 
     set +e
-    for home in "${BSD_HOME[@]}"; do
+    for mirror in "${MIRRORS[@]}"; do
         # we need a timeout in case the server is down to avoid
         # infinitely hanging. timeout error code is always 124
         # these mirrors can be quite slow, so have a long timeout
-        timeout 20s curl --silent --list-only --location "${home}/${BSD_ARCH}/" >/dev/null
+        timeout 20s curl --retry 3 -sSflL "${mirror}/${FREEBSD_ARCH}/" >/dev/null
```

Comment:
So as of very recently, this no longer works unless we add `http` to the front:

```bash
$ mirror="ftp.freebsd.org/pub/FreeBSD/releases"
$ url="${mirror}/amd64"
$ curl --silent --list-only --location "${url}"
CI-IMAGES
ISO-IMAGES
README.TXT
TIMESTAMP
VM-IMAGES
amd64
arm
arm64
i386
powerpc
riscv
sparc64
```

This is because the "file" no longer exists, which `--list-only` is masking. We should *always** use the HTTP gateway for this (and probably everything else).